JustSaying.IntegrationTests/Fluent/MessagingBusBuilderTests.cs

```diff
@@ -0,0 +1,271 @@
+                               .Services((options) => options.WithMessageMonitoring(() => new MyMonitor()));
+                    })
+                .AddJustSayingHandler<QueueMessage, QueueHandler>();
+
+            IServiceProvider serviceProvider = services.BuildServiceProvider();
+
+            IMessagePublisher publisher = serviceProvider.GetRequiredService<IMessagePublisher>();
+            IMessagingBus listener = serviceProvider.GetRequiredService<IMessagingBus>();
+
+            using (var source = new CancellationTokenSource(TimeSpan.FromSeconds(20)))
+            {
+                // Act
+                listener.Start(source.Token);
+
+                var message = new QueueMessage();
+
+                await publisher.PublishAsync(message, source.Token);
+
+                // Assert
+                while (!source.IsCancellationRequested && !QueueHandler.MessageIds.Contains(message.Id))
+                {
+                    await Task.Delay(TimeSpan.FromSeconds(1), source.Token);
+                }
+
+                QueueHandler.MessageIds.ShouldContain(message.Id);
+            }
+        }
+
+        [AwsFact]
+        public async Task Can_Create_Messaging_Bus_Fluently_For_A_Topic()
+        {
+            // Arrange
+            var services = new ServiceCollection()
+                .AddLogging((p) => p.AddXUnit(OutputHelper))
+                .AddJustSaying(
+                    (builder) =>
+                    {
+                        builder.Client((options) => options.WithBasicCredentials("accessKey", "secretKey").WithServiceUri(TestEnvironment.SimulatorUrl))
+                               .Messaging((options) => options.WithRegions("eu-west-1"))
+                               .Publications((options) => options.WithTopic<TopicMessage>())
+                               .Subscriptions((options) => options.ForTopic<TopicMessage>());
+                    })
+                .AddJustSayingHandler<TopicMessage, TopicHandler>();
+
+            IServiceProvider serviceProvider = services.BuildServiceProvider();
+
+            IMessagePublisher publisher = serviceProvider.GetRequiredService<IMessagePublisher>();
+            IMessagingBus listener = serviceProvider.GetRequiredService<IMessagingBus>();
+
+            using (var source = new CancellationTokenSource(TimeSpan.FromSeconds(20)))
+            {
+                // Act
+                listener.Start(source.Token);
+
+                var message = new TopicMessage();
+
+                await publisher.PublishAsync(message, source.Token);
+
+                // Assert
+                while (!source.IsCancellationRequested && !TopicHandler.MessageIds.Contains(message.Id))
+                {
+                    await Task.Delay(TimeSpan.FromSeconds(1), source.Token);
+                }
+
+                TopicHandler.MessageIds.ShouldContain(message.Id);
+            }
+        }
```

Comment:
This test here is what it's about 👏 The api here looks much more approachable than v6